packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -321,31 +321,31 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, ICrossChainSync {
     function _calcBasefee(
         EIP1559Config memory config,
         uint256 timeSinceParent,
-        uint32 gasLimit,
         uint32 parentGasUsed
     )
         private
         view
         returns (uint256 _basefee, uint64 _gasExcess)
     {
-        // Very important to cap _gasExcess uint64
         unchecked {
             uint32 parentGasUsedNet;
             if (parentGasUsed > LibL2Consts.ANCHOR_GAS_COST) {
                 parentGasUsedNet = parentGasUsed - LibL2Consts.ANCHOR_GAS_COST;
             }
 
-            uint256 a = uint256(gasExcess) + parentGasUsedNet;
-            uint256 b = config.gasIssuedPerSecond * timeSinceParent;
-            _gasExcess = uint64((a.max(b) - b).min(type(uint64).max));
+            uint256 issued = timeSinceParent * config.gasIssuedPerSecond;
+            uint256 excess = (uint256(gasExcess) + parentGasUsedNet).max(issued);
+            // Very important to cap _gasExcess uint64
+            _gasExcess = uint64((excess - issued).min(type(uint64).max));
         }
 
         _basefee = Lib1559Math.calculatePrice({
             xscale: config.xscale,
             yscale: config.yscale,
             xExcess: _gasExcess,
-            xPurchase: gasLimit
+            xPurchase: 0
```

Comment:
This will now approximate the derivative by calculating the difference between two points with a step of `1`. Theoretically should be okay, but I don't know how this will effect the accuracy because 1 is a tiny difference and there are approximations being done.

Reply:
Feel free to optimize it further.